Libraries/PeriphDrivers/Source/SPI/spi_reva1.c

```diff
@@ -1296,22 +1297,25 @@ int MXC_SPI_RevA1_SlaveTransactionDMA(mxc_spi_reva_req_t *req, int reqselTx, int
 void MXC_SPI_RevA1_DMACallback(int ch, int error)
 {
     mxc_spi_reva_req_t *temp_req;
+    uint8_t req_done;
 
     for (int i = 0; i < MXC_SPI_INSTANCES; i++) {
         if (states[i].req != NULL) {
             if (states[i].channelTx == ch) {
-                states[i].req_done++;
+                req_done = atomic_fetch_add(&states[i].req_done, 1);
             } else if (states[i].channelRx == ch) {
-                states[i].req_done++;
+                req_done = atomic_fetch_add(&states[i].req_done, 1);
                 //save the request
                 temp_req = states[i].req;
 
                 if (MXC_SPI_GetDataSize((mxc_spi_regs_t *)temp_req->spi) > 8) {
                     MXC_SPI_RevA1_SwapByte(temp_req->rxData, temp_req->rxLen);
                 }
+            } else {
+                continue;
             }
 
-            if (!states[i].txrx_req || (states[i].txrx_req && states[i].req_done == 2)) {
+            if (!states[i].txrx_req || (states[i].txrx_req && req_done == 1)) {
```

Comment:
Might be worth adding a comment saying something like:
```
// Compare req_done to '1' instead of '2' because 'req_done' holds the previous value of 'states[i].req_done',
//  and not the updated value incremented earlier with 'atomic_fetch_add(...)'.
```

Reply:
@JeonChangmin Aren't `states[i].req_done == 2` and `req_done == 1` logically equivalent after the `atomic_fetch_add`?

I think checking `states[i].req_done == 2` is a little more immediately readable against the rest of the driver design